### PR TITLE
Document how to run integration tests locally

### PR DESCRIPTION
**Pull request by greg0ire**
I am also replacing the commented out connection parameters in
phpunit.xml.dist with a link to this paragraph because they have
diverged from the one we use in the CI since #4067.

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -22,24 +22,11 @@
...
         -->
-        <!--<var name="db_event_subscribers" value="Doctrine\DBAL\Event\Listeners\OracleSessionInit">-->
-
-        <!-- Privileged user connection parameters. Used to create and drop the test database -->
-        <var name="tmpdb_driver" value="pdo_mysql"/>
-        <var name="tmpdb_host" value="localhost" />
-        <var name="tmpdb_port" value="3306"/>
-        <var name="tmpdb_user" value="root" />
-        <var name="tmpdb_password" value="" />
-        <var name="tmpdb_dbname" value="doctrine_tests_tmp" />
```
Comment (greg0ire) [inline phpunit.xml.dist:42]: Another reason for this removal is a discussion with  @alli83 about how to run the test suite, where I recommended either using a CI file or these instructions. She reported that the latter did not work (and the former works :slightly_smiling_face: ).

**Review by derrabus — DISMISSED** [review]
> We do not currently have a recommendation on how to spin up a database
server.

We could recommend Docker though.

**Comment by greg0ire** [comment]
@derrabus if I'm honest, using Docker is exactly how I would do it, so I think it's fair to recommend it indeed. I also removed a previous sentence that felt out of place and replaced it with another one about versions.

File: docs/en/reference/testing.rst [hunk 2]
```diff
@@ -62,6 +62,33 @@ multiple concurrent database connections, transactions, locking, performance-rel
...
+
+    $ phpunit -c ci/github/pdo_mysql.xml
+
+We do not currently have specific instructions on how to run a Database
+server, but we do recommend Docker as a convenient way to do so.
+To know which version of a database you should use, you can check
+``.github/workflows/continuous-integration.yml`` to see which versions
+we officially support. For instance, the
+supported PostgreSQL versions are listed under
+``jobs.phpunit-postgres.strategy.matrix.postgres-version``.
```
Comment (derrabus) [inline docs/en/reference/testing.rst]: _Supported_ versions are listed here: https://www.doctrine-project.org/projects/doctrine-dbal/en/4.0/reference/platforms.html

The CI config tells you what versions we _test against_.

Comment (greg0ire) [reply docs/en/reference/testing.rst]: Addressed.

**Review by SenseException — DISMISSED** [review]

File: docs/en/reference/testing.rst [hunk 3]
```diff
@@ -62,6 +62,31 @@ multiple concurrent database connections, transactions, locking, performance-rel
...
+
+.. code-block:: console
+
+    $ phpunit -c ci/github/pdo_mysql.xml
+
+We do not currently have specific instructions on how to run a Database
+server, but we do recommend Docker as a convenient way to do so.
+We do not recommend running against a particular version of the chosen
+RDBMS either, as long as you pick one of the `officially supported versions
+<https://www.doctrine-project.org/projects/doctrine-dbal/en/4.0/reference/platforms.html>`_.
```
Comment (derrabus) [inline docs/en/reference/testing.rst]: Shouldn't this be an internal link that should be resolved during rendering?

Comment (greg0ire) [reply docs/en/reference/testing.rst]: Oh right, definitely!

**Review by derrabus — APPROVED** [review]

**Pull request merged**